gdrive

```diff
@@ -84,53 +80,63 @@ def search_filename(service, file_name=None, page_size=10, list_all=False, page_
         query = "name contains '%s'" % file_name
     if page_token is None:
         page_token = ''
-    fields = 'nextPageToken, files/id, files/name, files/size, files/modifiedTime, files/owners'
+    fields = ('nextPageToken, files/id, files/name, files/size, '
+              'files/modifiedTime, files/modifiedByMeTime, files/owners')
     search = service.files().list(q=query,
-                                  orderBy='modifiedTime desc',
+                                  orderBy='modifiedByMeTime desc',
                                   fields=fields,
                                   pageSize=page_size,
                                   pageToken=page_token).execute()
     return search.get('files', []), search.get('nextPageToken', None)
 
 def list_files(service, file_name, list_all=False):
     """List files present on user's Google Drive"""
-    files, next_page = search_filename(service, file_name, 10, list_all)
-    for file in files:
-        describe_file(file)
+    files_found, next_page = search_filename(service, file_name, 10, list_all)
+    for metadata in files_found:
+        describe_file(metadata)
     while next_page is not None:
         if input(">>> Press Enter for the next results\n"):
             break
-        files, next_page = search_filename(service, file_name, 10, next_page)
-        for file in files:
-            describe_file(file)
+        files_found, next_page = search_filename(service, file_name, 10, list_all, next_page)
+        for metadata in files_found:
+            describe_file(metadata)
 
 def describe_file(file):
     """Print file metadata"""
     print("Name: %s" % file["name"])
+    print("Owner: %s" % file["owners"][0]["displayName"])
     if 'size' in file:
         print("Size: %s" % size_to_human_readable(float(file["size"])))
-    print("Modified: %s" % file["modifiedTime"])
-    print("Owner: %s" % file["owners"][0]["displayName"])
+    if 'modifiedByMeTime' in file:
+        print("Modified by Me: %s" % file["modifiedByMeTime"])
+    else:
+        print("Modified: %s" % file["modifiedTime"])
     print("ID: %s\n" % file["id"])
 
-def download_name(service, file_name):
+def download_name(service, filename, list_all=False):
     """Download a file by name"""
-    if file_name is None:
+    if not list_all and filename is None:
         print('Please inform the name of the file you want to download. Exiting...')
-        return
-    file_found, _ = search_filename(service, file_name, 1)
-    if not file_found:
-        return print("Could not find any file with the name '%s'" % file_name)
-    describe_file(file_found[0])
-    download(service, file_found[0]["id"])
-
-def download(service, file_id):
+        return None
+    files_found, _ = search_filename(service, filename, 1, list_all)
+    if not files_found:
+        print("Could not find any file with the name '%s'" % filename)
```

Comment:
“With name” is proper English?

Reply:
Changed to reflect current state of the function
"Could not find any file that contains '%s' on the name"